src/main/java/com/faforever/api/permission/PermissionService.java

```diff
@@ -0,0 +1,67 @@
+package com.faforever.api.permission;
+
+import com.faforever.api.data.domain.Permission;
+import com.faforever.api.data.domain.Role;
+import com.faforever.api.data.domain.RolePermissionAssignment;
+import com.faforever.api.data.domain.RoleUserAssignment;
+import com.faforever.api.data.domain.User;
+import com.faforever.api.user.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+@Service
+public class PermissionService {
+  private final PermissionRepository permissionRepository;
+  private final RoleRepository roleRepository;
+  private final RoleUserAssignmentRepository roleUserAssignmentRepository;
+  private final UserRepository userRepository;
+
+  public PermissionService(PermissionRepository permissionRepository, RoleRepository roleRepository, RoleUserAssignmentRepository roleUserAssignmentRepository, UserRepository userRepository) {
+    this.permissionRepository = permissionRepository;
+    this.roleRepository = roleRepository;
+    this.roleUserAssignmentRepository = roleUserAssignmentRepository;
+    this.userRepository = userRepository;
+  }
+
+  public Permission createPermission(String name) {
+    Permission permission = new Permission().setName(name);
+    permissionRepository.save(permission);
+    return permission;
+  }
+
+  public Role createRole(String name, Permission... permissions) {
+    Role role = new Role().setName(name);
+    Arrays.stream(permissions).forEach(permission -> assignPermissionToRole(permission, role));
+    roleRepository.save(role);
+    return role;
```

Comment:
`return roleRepository.save(role)` because it may actually return a different object.